### PR TITLE
Add Cloud Spanner DML/PDML samples.

### PR DESCRIPTION
**Pull request by jsimonweb**
Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/dml.js [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+    transaction
+      .runUpdate({
+        sql: `UPDATE Singers SET LastName = 'Grant'
+        WHERE STRUCT<FirstName STRING, LastName STRING>(FirstName, LastName) = @name`,
+        params: {
+          name: nameStruct,
+        },
+      })
+      .then(rowCount => {
+        console.log(`Successfully updated ${rowCount} record.`);
```
Comment (crwilcox) [inline samples/dml.js]: should this be `records.`? Since record is correct only for 1==rowCount?

File: samples/dml.js [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+      console.error(err);
+      return;
+    }
+    transaction
+      .runUpdate({
+        sql: `INSERT Singers (SingerId, FirstName, LastName)
+          VALUES (11, 'Timothy', 'Campbell')`,
+      })
+      // Queries rows from the Singers table
+      .run({sql: `SELECT FirstName, LastName FROM Singers`})
```
Comment (callmehiphop) [inline samples/dml.js]: This needs to be 

```js
.then(() => transaction.run('SELECT FirstName, LastName FROM Singers'))`
```

File: samples/dml.js [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+        secondBudget += transferAmount;
+        firstBudget -= transferAmount;
+
+        console.log(firstBudget, secondBudget);
+
+        // Update the database
+        // Note: Cloud Spanner interprets Node.js numbers as FLOAT64s, so they
+        // must be converted (back) to strings before being inserted as INT64s.
+
+        transaction.runUpdate({
```
Comment (callmehiphop) [inline samples/dml.js]: Do we want a `return` or `Promise.all()` for these two calls?

Comment (jsimonweb) [reply samples/dml.js]: Leaving as is for now as this follows the pattern in transactions.js

File: samples/dml.js (hunk 1 above)
Comment (jsimonweb) [reply samples/dml.js]: This is intended as a single record update.

File: samples/dml.js (hunk 3 above)
Comment (callmehiphop) [reply samples/dml.js]: I might be reading it wrong, but it appears to run two concurrent updates and the promise chain does not wait for either to finish before committing the transaction. The closest I could find in transaction.js to this was the [usage of `update`](https://github.com/googleapis/nodejs-spanner/blob/master/samples/transaction.js#L176) which is a synchronous method whereas `runUpdate` is async.

**Pull request merged**